cqf-fhir-utility/src/main/java/org/opencds/cqf/fhir/utility/visitor/WithdrawVisitor.java

```diff
@@ -0,0 +1,66 @@
+package org.opencds.cqf.fhir.utility.visitor;
+
+import ca.uhn.fhir.rest.server.exceptions.PreconditionFailedException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.hl7.fhir.instance.model.api.IBase;
+import org.hl7.fhir.instance.model.api.IBaseParameters;
+import org.hl7.fhir.instance.model.api.IDomainResource;
+import org.opencds.cqf.fhir.api.Repository;
+import org.opencds.cqf.fhir.utility.BundleHelper;
+import org.opencds.cqf.fhir.utility.PackageHelper;
+import org.opencds.cqf.fhir.utility.adapter.KnowledgeArtifactAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WithdrawVisitor implements KnowledgeArtifactVisitor {
+    private Logger log = LoggerFactory.getLogger(WithdrawVisitor.class);
+    String isOwnedUrl = "http://hl7.org/fhir/StructureDefinition/crmi-isOwned";
+
+    @Override
+    public IBase visit(
+            KnowledgeArtifactAdapter rootAdapter, Repository repository, IBaseParameters operationParameters) {
+        if (!rootAdapter.getStatus().equals("draft")) {
+            throw new PreconditionFailedException("Cannot withdraw an artifact that is not in draft status");
+        }
+        var resToUpdate = new ArrayList<IDomainResource>();
+        resToUpdate.add(rootAdapter.get());
+
+        var resourcesToUpdate = gatherComposedOfChildren(rootAdapter, repository, resToUpdate);
+
+        var fhirVersion = rootAdapter.get().getStructureFhirVersionEnum();
+
+        var transactionBundle = BundleHelper.newBundle(fhirVersion, null, "transaction");
+        for (var artifact : resourcesToUpdate) {
+            var entry = PackageHelper.deleteEntry(artifact);
+            BundleHelper.addEntry(transactionBundle, entry);
+        }
+
+        return repository.transaction(transactionBundle);
+    }
+
+    private List<IDomainResource> gatherComposedOfChildren(
```

Comment:
NITPICK: Calling this getComponents might be more appropriate and consistent with what we've done elsewhere. "Component" is the concept that implies a child via "composed-of" reference.

Reply:
fixed :)